.github/workflows/migration.yml

```diff
@@ -33,17 +33,16 @@ jobs:
           docker restart ${COMPOSE_PROJECT_NAME}-mysql-1 && sleep 10
       - name: Build test image
         env:
-          BUILD_IMAGE: mattermostdevelopment/mattermost-build-server:${{ steps.go.outputs.GO_VERSION }}
+          BUILD_IMAGE: golang:${{ steps.go.outputs.GO_VERSION }}-bookworm
+          PGLOADER_IMAGE: mattermost/mattermost-pgloader
         run: |
-          echo -e "FROM $BUILD_IMAGE\n\nRUN apt update && \
-            apt remove -y pgloader && \
-            apt install -y sbcl unzip libsqlite3-dev make curl gawk freetds-dev libzip-dev && \
-            git clone https://github.com/dimitri/pgloader.git && \
-            cd pgloader && \
-            git checkout \$(git tag --list --sort=-v:refname "v*" | head -n1) && \
-            make && \
-            mv build/bin/pgloader /bin" >> dockerfile_tmp
-          docker build -t $TEST_IMAGE - < dockerfile_tmp
+          docker build -t $TEST_IMAGE - <<EOF
+          FROM $PGLOADER_IMAGE AS pgloader
+          FROM $BUILD_IMAGE
+          COPY --from=pgloader /usr/bin/pgloader /usr/bin/pgloader
+          EOF
+          # Print out installed pgloader version
+          docker run --rm $TEST_IMAGE pgloader --version
```

Comment:
Instead of doing docker build every time, how about adding it directly to the `mattermostdevelopment/mattermost-build-server` image - https://github.com/mattermost/mattermost/blob/master/server/build/Dockerfile.buildenv#L3?

Reply:
I 100% agree with the idea, but this depends on our ability to upgrade our build image from Debian Bullseye to Bookworm, otherwise the `pgloader` binary fails executing (pgloader is linked against newer GLIBC versions than the ones in Bullseye)

cc @agnivade would it be acceptable to bump the [build image](https://github.com/mattermost/mattermost/blob/master/server/build/Dockerfile.buildenv) from `bullseye` to `bookworm`? I think that's ok, and closer to our actual MM server image, which is built on Ubuntu Noble

Reply:
Should be fine. I'll defer to you/ @toninis really.

Reply:
Ok, thanks for the feedback. I'll take a second look at the server build later, and see if our minimum supported OS version would still be supported if we were to do this change :+1:

Reply:
I had a look today and it's safer to not upgrade the build image to Bookworm yet, because it utilizes GLIBC 2.36, which is newer than our oldest supported Ubuntu's GLIBC (which is 2.31, on Focal).
```
$ docker run --rm -it ubuntu:focal dpkg --list  libc6 | grep libc6
ii  libc6:amd64    2.31-0ubuntu9.16 amd64        GNU C Library: Shared libraries

$ docker run --rm -it debian:bookworm dpkg --list  libc6 | grep libc6
ii  libc6:amd64    2.36-9       amd64        GNU C Library: Shared libraries
```
If we compile the server in Bookworm, we may not be able to run it on Focal anymore: let's tackle this when we stop supporting Ubuntu Focal.